### DISCUSSION TITLE
Line Search Options

### DISCUSSION THREAD

**Discussion by maxnezdyur** [idaholab/moose#18993] [Q&A Modules: General] [answered]
I apologize if I missed it somewhere.  I am trying to figure out the different line search options. The most information I could find was this "Options: basic, bt, contact, cp, default, l2, none, project, shell". The problem I am having is that I am not sure which line search option is which. I can assume bt is backtracking, but I don't know what the others are. Is there a resource I missed when looking at the different line search options?

**Comment by GiudGiud** [comment, marked as answer]
Hello

There's a documentation page but it's super short and I m not sure where it is online so no need to look for it. 
The line searches are mostly inherited from Petsc. You should look at petsc documentation for more explanation on bt etc
If they dont work well, try line_search = 'none'

Guillaume

**Reply by GiudGiud** [reply]
Can you post a link once you've found the petsc documentation? I'll upgrade our line search docs in moose
I created an issue https://github.com/idaholab/moose/issues/19006

**Reply by maxnezdyur** [reply]
This is good to know. Thank you for the information. https://petsc.org/release/docs/manualpages/SNES/SNESLineSearchSetType.html
I am sure this is the petsc documentation you are asking for; it includes references to different line searches with many of the same names in Moose.

**Reply by GiudGiud** [reply]
Thank you!

**Reply by hugary1995** [reply]
@GiudGiud while you are at it, maybe it's helpful to also link to the documentation for [the contact line search](https://mooseframework.inl.gov/moose/source/linesearches/ContactLineSearch.html).
